File: greedy/1784_check_if_binary_string_has_at_most_one_segment_of_ones/1784_check_if_binary_string_has_at_most_one_segment_of_ones.go
package main

import (
	"fmt"
	"strings"
)

func checkOnesSegment(s string) bool {
	return !strings.Contains(s, "01")
}

func main() {
	s := "110"
	res := checkOnesSegment(s)
	fmt.Println(res)
}

File: greedy/1784_check_if_binary_string_has_at_most_one_segment_of_ones/1784_check_if_binary_string_has_at_most_one_segment_of_ones.py
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        new = 0
        for i in s:
            if i == '0':
                new = 1
            else:
                if new == 1:
                    return False
        return True


s = '1001'
res = Solution().checkOnesSegment(s)
print(res)